src/framework/theme/components/stepper/stepper.component.ts

```diff
@@ -176,6 +183,12 @@ export class NbStepperComponent {
   }
   protected _linear = true;
 
+  /**
+   * Emits when step changed
+   * @type {EventEmitter<number>}
+   */
+  @Output() stepChanged = new EventEmitter<number>();
```

Comment:
Sorry, last time I completely missed out that `stepChanged` is emitting only step index. I think we should provide more info as a step itself and also a previous step and its index. So the event interface should be:
```
interface NbStepChangedEvent {
  index: number;
  step: NbStepComponent;
  previouslySelectedIndex: number;
  previouslySelectedStep: NbStepComponent;
}
```
And then please update event description in documentation
>This event emits current step index.

Reply:
The problem here, that after the user clicks on some step, and then click prev we should add additional logic to control previouslySelectedIndex become selectedIndex, and current will be -1, so I discard logic from Set, and move it to prev next and reset functions. Is it ok?